app/models/communitarian/resolution.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require_relative "../../jobs/regular/reopen_resolution"
+require_relative "../post"
+
+module Communitarian
+  class Resolution
+    using Communitarian
+
+    def self.reopened_poll_post_attributes
+      %w[user_id topic_id raw post_type last_editor_id last_version_at user_deleted public_version]
+    end
+
+    def self.reopen_weekly_resolution!(original_post)
+      original_poll = original_post.polls.first
+      return if original_post.user_deleted || !original_post.resolution?
```

Comment:
maybe put check `return if original_post.user_deleted || !original_post.resolution?` higher in the method?